### DISCUSSION TITLE
Performance comparison

### DISCUSSION THREAD

**Discussion by meandmytram** [tencent-quantum-lab/tensorcircuit#3] [General]
Hey guys, amazing library! Just out of curiosity, was there any benchmarking done against other TN packages like `tensornetwork` or `quimb`?

**Comment by refraction-ray** [comment]
Thanks for your good question.

TensorCircuit is built on top of ``TensorNetwork`` package with built-in differentiable circuit class and some hacks on the tensor network contraction path finder (TensorCircuit supports``cotengra`` package as the path finder). 
Therefore, TensorCircuit is definitely the choice over TensorNetwork package in terms of quantum circuit simulation with its handy API and better performance.

``quimb`` is another fantastic package from the same author for ``cotengra``. In terms of API design, we believe that TensorCircuit has better ``Circuit`` object with more built-in functionalities since our package is more focused on circuit simulation while ``quimb`` is a more general tensor network package. In terms of performance, careful benchmarks are required and we are on the way. For now, what I can say is that TensorCircuit package focuses more on end-to-end ``jit`` and ``vmap`` paradigms, which may bring performance advantages compared to ``quimb``.
